assets/src/design-system/components/tooltip/index.js

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { useState, useRef, useMemo, useCallback } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { Popup, PLACEMENT } from '../popup';
+import { prettifyShortcut } from '../keyboard';
+import { THEME_CONSTANTS } from '../../theme';
+import { Text } from '../typography';
+import { SvgForTail, Tail, SVG_TOOLTIP_TAIL_ID } from './tail';
+
+const SPACE_BETWEEN_TOOLTIP_AND_ELEMENT = 8;
+
+const Wrapper = styled.div`
+  position: relative;
+`;
+
+const Tooltip = styled.div`
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  text-align: center;
+  flex-direction: row;
+  max-width: 13em;
+  transition: 0.4s opacity;
+  opacity: ${({ shown }) => (shown ? 1 : 0)};
+  pointer-events: ${({ shown }) => (shown ? 'all' : 'none')};
+  z-index: 9999;
+  border-radius: 4px;
+  background-color: ${({ theme }) => theme.colors.inverted.bg.primary};
+`;
+
+const TooltipText = styled(Text)`
+  color: ${({ theme }) => theme.colors.inverted.fg.primary};
+  padding: 10px;
+`;
+
+const getBoundingBoxCenter = ({ x, width }) => x + width / 2;
+function WithTooltip({
```

Comment:
This is a nit, but I think a component that starts with a verb is confusing. That to me more indicates a hook or HOC.

Reply:
totally fair! i was originally thinking this would keep it in line with the editor (this is what it's called there) but we have a fresh start, we should improve continuity in naming. I'll go ahead and update this.